packages/backend-app-api/src/services/implementations/httpRouter/httpRouterServiceFactory.ts

```diff
@@ -52,8 +71,14 @@ export const httpRouterServiceFactory = createServiceFactory(
 
       return {
```

Comment:
This could be a good spot to make the new authentication behavior opt-in. You have access to the config you could introduce a setting (`backend.enforceAuthentication`?) that defaults to `false` and return either the original `router.use(handler)` implementation or the new `router.use(authenticate, handler)` based on the value of this setting. Users can then opt-in in their config and after x versions you could remove the setting or make it opt-out instead by defaulting to `true`.

You'd have to introduce a similar check in the TechDocs plugin because you use `useWithCookieAuthentication` there instead of `use`.

Reply:
I like the idea! I think the config would flip the authentication handler on or off so it could work with both.

Reply:
Might also be a good idea to allow it to be configurable which plugins/endpoints you want to selectively enable/disable auth. In the current recommended [setup](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md#new-backend-system-setup) you're able to create a whitelist of non-authed plugins but this will now lock it down to whether each plugin is implemented to call `httpRouter.useWithoutAuthentication` or `httpRouter.use`. A Backstage integrator may wish to have specific plugins be publicly accessible and stray from the default enforcement. Anexample is that the [badges plugin](https://github.com/backstage/backstage/tree/master/plugins/badges-backend) requires certain routes to be publicly accessible so that the badge will render correctly on external sites

Reply:
@kuangp Can you share some use cases where the default plugin auth wouldn't be sufficient? I _think_ the current solution where the plugins provide different levels of authentication would fulfill the example in that guide (TechDocs + `/api/auth/*`). Thanks for bringing up badges, I still have to adjust the `app-backend` package and badges would be good to update as well.

The other thing that comes to mind would be guests/public users which might be solved by #22565?

Reply:
Don't have a specific use case in mind - just a suggestion that it could be a useful option to allow the integrator to disable auth on a per plugin/endpoint basis if needed and is somewhat analogous to how integrators are able to customize permissions per plugin

Reply:
I could see adding a config value with an allowlist to get through the protected-by-default layer. That might be relevant if they interact with a plugin that hasn't correctly updated to support authentication by default? If the integrator is writing their own plugins, they could separate their routers into an authenticated router and an unauthenticated one, that's the approach I'll take for badges.